src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -61,8 +62,12 @@ internal static Exception CreateOpenSslCryptographicException()
                 return new CryptographicException();
             }
 
+            // Even though ErrGetError returns ulong (C++ unsigned long), we 
+            // really only expect error codes in the UInt32 range
+            Debug.Assert(error <= uint.MaxValue, "ErrGetError should only return error codes in the UInt32 range.");
+
             // Inline version of the ERR_GET_REASON macro.
-            uint reason = error & ReasonMask;
+            ulong reason = error & ReasonMask;
```

Comment:
Where is ERR_R_MALLOC_FAILURE below defined? If it's a constant in C#, there should be a static assert in shim that we have the value right.